cwd/app.js

```diff
@@ -49,20 +51,23 @@
           var index = arr.indexOf(document.getElementById(this.id));
 
           arr[index].style.display = "none";
+
           if (index != arr.length - 1) {
             arr[index+1].style.display = "block";
             renderView(views[index+1]);
+            if(window.location.hash)
+              window.location.hash = views[index+1].name;
 
           } else {
             arr[0].style.display = "block";
```

Comment:
I'm realizing what I actually want is for `else` to be `else if (arr.length > 0)`.  That will cause this not to error when there are 0 views or if the view buttons do not use glow on hover or something.

Given that `arr`, `views`, and `hashes` are all linked in terms of their length and indexes, why not make them truly linked?  Consider an array of objects like:
`{ element: arr[0], view: views[0], hash: hash[0] }`
It will make the different variables more understandable.

Reply:
I think the easiest way to form this object is to make the `views` variable map to a new object that stores the glyph object, DOM element, and hash.